pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
+										},
+									},
+								},
+								Lifecycle: &corev1.Lifecycle{
+									PreStop: &corev1.Handler{
+										Exec: &corev1.ExecAction{
+											Command: []string{
+												"/bin/sh", "-c", "PID=$(pgrep -x scylla);supervisorctl stop scylla; while kill -0 $PID; do sleep 1; done;",
+											},
+										},
+									},
+								},
+							},
+							{
+								Name:            "scylla-manager-agent",
+								Image:           ":",
+								ImagePullPolicy: corev1.PullIfNotPresent,
+								Args: []string{
+									"-c",
+									naming.ScyllaAgentConfigDefaultFile,
+									"-c",
+									path.Join(naming.ScyllaAgentConfigDirName, naming.ScyllaAgentConfigFileName),
+									"-c",
+									path.Join(naming.ScyllaAgentConfigDirName, naming.ScyllaAgentAuthTokenFileName),
+								},
+								Ports: []corev1.ContainerPort{
+									{
+										Name:          "agent-rest-api",
+										ContainerPort: 10001,
+									},
+								},
+								VolumeMounts: []corev1.VolumeMount{
+									{
+										Name:      naming.PVCTemplateName,
+										MountPath: naming.DataDir,
+									},
+									{
+										Name:      scyllaAgentConfigVolumeName,
+										MountPath: path.Join(naming.ScyllaAgentConfigDirName, naming.ScyllaAgentConfigFileName),
+										SubPath:   naming.ScyllaAgentConfigFileName,
+										ReadOnly:  true,
+									},
+									{
+										Name:      scyllaAgentAuthTokenVolumeName,
+										MountPath: path.Join(naming.ScyllaAgentConfigDirName, naming.ScyllaAgentAuthTokenFileName),
+										SubPath:   naming.ScyllaAgentAuthTokenFileName,
+										ReadOnly:  true,
+									},
+								},
+								Resources: basicRack.Resources,
+							},
+						},
+						DNSPolicy:          "ClusterFirstWithHostNet",
+						ServiceAccountName: "basic-member",
+						Affinity:           &corev1.Affinity{},
+					},
+				},
+				VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:   naming.PVCTemplateName,
+							Labels: newBasicStatefulSetLabels(),
+						},
+						Spec: corev1.PersistentVolumeClaimSpec{
+							AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+							StorageClassName: basicRack.Storage.StorageClassName,
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceStorage: resource.MustParse(basicRack.Storage.Capacity),
+								},
+							},
+						},
+					},
+				},
+				ServiceName:         naming.HeadlessServiceNameForCluster(newBasicScyllaCluster()),
+				PodManagementPolicy: appsv1.OrderedReadyPodManagement,
+				UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+					Type: appsv1.RollingUpdateStatefulSetStrategyType,
+					RollingUpdate: &appsv1.RollingUpdateStatefulSetStrategy{
+						Partition: pointer.Int32Ptr(0),
+					},
+				},
+			},
+		}
+	}
+
+	newNonNilImagePullSecrets := func() []corev1.LocalObjectReference {
```

Comment:
nit: I'd find it more readable we inline short functions like that, but that might be opinionated. in the other case the nonNil prefix is kind of implicit, I'd not assume you make a function to create nil

Reply:
I was taking example from how it's done in existing test, e.g. [cluster_validation_test](https://github.com/scylladb/scylla-operator/blob/603327d94c3ee0b58d3dc46fa8260588f003fb97/pkg/api/validation/cluster_validation_test.go#L30). Inlining would require duplicating the anonymous function in each test case. What would you change the "nonNil" to?

Reply:
Thanks, I changed the shorter functions to inline and renamed the other ones